hibernate-core/src/main/java/org/hibernate/engine/spi/SessionDelegatorBaseImpl.java

```diff
@@ -1163,7 +1163,7 @@ public Integer getJdbcBatchSize() {
 	}
 
 	@Override
-	public void setJdbcBatchSize(int jdbcBatchSize) {
+	public void setJdbcBatchSize(Integer jdbcBatchSize) {
```

Comment:
I just wonder if the argument should be an Optional

On Mon, Jun 13, 2016 at 1:51 PM Vlad Mihalcea <notifications@github.com>
wrote:

> In
> hibernate-core/src/main/java/org/hibernate/engine/spi/SessionDelegatorBaseImpl.java
> <https://github.com/hibernate/hibernate-orm/pull/1423#discussion_r66846227>
> :
>
> > @@ -1163,7 +1163,7 @@ public Integer getJdbcBatchSize() {
> >  	}
> >
> >  	@Override
> > -	public void setJdbcBatchSize(int jdbcBatchSize) {
> > +	public void setJdbcBatchSize(Integer jdbcBatchSize) {
>
> There are basically two use cases that can benefit from this change:
>
>    1. When using an EXTENDED Persistence Context, we might have only a
>    single service layer method override the default JDBC batch size. When the
>    call gets out of this service method, we want to reset the Session-level
>    JDBC setting to null since other subsequent service layer methods
>    might want to use the SessionFactory setting.
>    2. Even in the scope of a single Session we might want that only a
>    bunch of entities use a given batch size, while the rest use the default
>    SessionFactory setting.
>
> If you don't like it, I can easily roll it back. Just let me know.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/hibernate/hibernate-orm/pull/1423/files/8513a55255a90ebbcf7a6501677975a71b7a8a73#r66846227>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAOUE6GmlMPi6nJi6BBbTtS4IhH8zMYzks5qLaaogaJpZM4I0dK9>
> .
>